app/models/draft_vaccination_record.rb

```diff
@@ -367,4 +368,22 @@ def delivery_site_matches_delivery_method
       errors.add(:delivery_site, :inclusion)
     end
   end
+
+  def patient_attended_session
+    if session&.requires_registration? &&
+         !PatientSession::RegistrationStatus
+           .for_patient_session(patient, session)
+           .allows_recording_vaccinations
+           .exists?
+      date = performed_at.to_date
+      errors.add(
+        :session_id,
```

Comment:
I'm wondering if it makes sense for this error to be attached to `:session_id`, I wonder if `:base` is more appropriate (since it's more like a general validation error for the model rather than a specific attribute).